pkg/config/model/viper.go

```diff
@@ -89,13 +90,18 @@ func (c *safeConfig) Set(key string, value interface{}, source Source) {
 		return
 	}
 
-	c.Lock()
-	defer c.Unlock()
-	c.configSources[source].Set(key, value)
-	c.mergeViperInstances(key)
+	// modify the config then release the lock to avoid deadlocks
+	var receivers []NotificationReceiver
+	func() {
+		c.Lock()
+		defer c.Unlock()
+		c.configSources[source].Set(key, value)
+		c.mergeViperInstances(key)
+		receivers = slices.Clone(c.notificationReceivers)
+	}()
```

Comment:
<!-- Nitpicks are small, trivial, but necessary changes. Distinguishing nitpick comments significantly helps direct the reader's attention to comments requiring more involvement.  -->
**🥜 nitpick:** is the closure only used to be able to `defer` directly ? I feel like it makes the code less clear than just `Unlocking` 3 lines down, what do you think ?‏

Reply:
Now that you mention it, I do think that's better, I will make that change.